File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    let answer = [0, 0];
    let endX = (board[0] - 1) / 2;
    let endY = (board[1] - 1) / 2;
    
    for (let i of keyinput) {
        switch(i){
            case 'left' : if(answer[0] != -endX) answer[0]--; break;
            case 'right' : if(answer[0] != endX) answer[0]++; break;
            case 'up' : if(answer[1] != endY) answer[1]++; break;
            case 'down' : if(answer[1] != -endY) answer[1]--; break;
        }
    }
    return answer;
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    answer = [0, 0]
    end_x = (board[0] - 1) // 2
    end_y = (board[1] - 1) // 2
    
    for command in keyinput:
        if command == 'left':
            if answer[0] > -end_x:
                answer[0] -= 1
        elif command == 'right':
            if answer[0] < end_x:
                answer[0] += 1
        elif command == 'up':
            if answer[1] < end_y:
                answer[1] += 1
        elif command == 'down':
            if answer[1] > -end_y:
                answer[1] -= 1        
    return answer